src/Pim/Component/Catalog/Factory/ProductValue/DateProductValueFactory.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Pim\Component\Catalog\Factory\ProductValue;
+
+use Akeneo\Component\StorageUtils\Exception\InvalidPropertyException;
+use Pim\Component\Catalog\Exception\InvalidArgumentException;
+use Pim\Component\Catalog\Model\AttributeInterface;
+
+/**
+ * Factory that creates date product values.
+ *
+ * @internal  Please, do not use this class directly. You must use \Pim\Component\Catalog\Factory\ProductValueFactory.
+ *
+ * @author    Damien Carcel (damien.carcel@akeneo.com)
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+class DateProductValueFactory implements ProductValueFactoryInterface
+{
+    /** @var string */
+    protected $dateProductValueClass;
+
+    /** @var string */
+    protected $supportedAttributeType;
+
+    /**
+     * @param string $dateProductValueClass
+     * @param string $supportedAttributeType
+     */
+    public function __construct($dateProductValueClass, $supportedAttributeType)
+    {
+        if (!class_exists($dateProductValueClass)) {
```

Comment:
imho it would be more interesting to check `... instanceof ProductValueInterface` here since we need its methods in `create()`.
Same comment for all the factories.